# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/ipc/ContentCompositorBridgeParent.cpp
# Commit: 71c122ac0ca7
# Full Hash: 71c122ac0ca73391866b1ef19f4f82bc2d28568b
# Author: David Parks <davidp99@gmail.com>
# Date: 2020-01-09 03:40:18
# Regressor Bug: 1477756
# File Overlap Count: 1
# Description:
#   Bug 1477756 - Initial out-of-process WebGL implementation. r=mccr8,handyman
#   
#   Splits WebGLContext into ClientWebGLContext and HostWebGLContext.  The Client enables the JS-control of a WebGL context in a content procecss while the Host executes the WebGL graphics operations (via a WebGLContext that maintains much of the existing code) in the compositor process.  At this point, the cross-process behavior is disabled -- this series of patches is an incremental step toward that final goal.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D54018
# ==============================================================================

diff -r 5cbf5bb11c58 -r 71c122ac0ca7 gfx/layers/ipc/ContentCompositorBridgeParent.cpp
--- a/gfx/layers/ipc/ContentCompositorBridgeParent.cpp	Wed Jan 08 21:52:03 2020 +0000
+++ b/gfx/layers/ipc/ContentCompositorBridgeParent.cpp	Wed Jan 08 22:19:14 2020 +0000
@@ -18,6 +18,9 @@
 #  include "mozilla/gfx/DeviceManagerDx.h"  // for DeviceManagerDx
 #  include "mozilla/layers/ImageDataSerializer.h"
 #endif
+#include "mozilla/dom/WebGLCrossProcessCommandQueue.h"
+#include "mozilla/dom/WebGLErrorQueue.h"
+#include "mozilla/dom/WebGLParent.h"
 #include "mozilla/ipc/Transport.h"           // for Transport
 #include "mozilla/layers/AnimationHelper.h"  // for CompositorAnimationStorage
 #include "mozilla/layers/APZCTreeManagerParent.h"  // for APZCTreeManagerParent
@@ -764,5 +767,18 @@
   return IPC_OK();
 }
 
+dom::PWebGLParent* ContentCompositorBridgeParent::AllocPWebGLParent(
+    const WebGLVersion& aVersion, HostWebGLCommandSink&& aCommandSink,
+    HostWebGLErrorSource&& aErrorSource) {
+  MOZ_ASSERT_UNREACHABLE("Unimplemented");
+  return nullptr;
+}
+
+bool ContentCompositorBridgeParent::DeallocPWebGLParent(
+    PWebGLParent* aWebGLParent) {
+  delete aWebGLParent;
+  return true;
+}
+
 }  // namespace layers
 }  // namespace mozilla